src/NServiceBus.Core/Licensing/LicenseSources.cs

```diff
@@ -21,30 +21,17 @@ public static LicenseSource[] GetLicenseSources(string licenseText, string licen
                 sources.Add(new LicenseSourceFilePath(licenseFilePath));
             }
 
-            sources.Add(new LicenseSourceConfigFile());
+            sources.Add(new LicenseSourceAppConfigLicenseSetting());
+            sources.Add(new LicenseSourceAppConfigLicensePathSetting());
 
-            sources.Add(new LicenseSourceFilePath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "NServiceBus", "License.xml")));
-            sources.Add(new LicenseSourceFilePath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "License", "License.xml")));
+            sources.Add(new LicenseSourceFilePath(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), "ParticularSoftware", "license.xml")));
+            sources.Add(new LicenseSourceFilePath(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "ParticularSoftware", "license.xml")));
+            sources.Add(new LicenseSourceFilePath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "license.xml")));
```

Comment:
Thinking through this a bit more, I'm now of the opinion that we shouldn't try to do anything special here at all.

For case-insensitive file systems, it won't matter how we case it here. If we added a second source here with a different casing, then we'd end up picking up the same license file twice.

Note: macOS is actually case-insensitive by default also, despite having unix underpinnings.

For case-sensitive file systems (linux), being aware of casing is simply the nature of the beast. If the exact casing doesn't exist, we could try to enumerate all of the files in the destination folder, and find a match ignoring case, but what if they've not cased that folder correctly either? How far up the entire path to the file do we try to accommodate mismatched casing?

At this point, I think we make sure we've got the exact casing correct in our documentation, and don't try to patch over this difference in platforms.